### ISSUE TITLE
El botón "+" no aparece en Yomichan

### ISSUE THREAD

**Issue by Patata099** [FooSoft/yomichan#2241]
No puedo agregar la mayor parte de las palabras que selecciono en Yomichan a Ankidroid desde hace dos días (el boton "+" no se muestra). De igual manera las palabras que ya tengo agregadas no muestran el pequeño libro que indica que ya fueron enviadas a Anki.

 La verdad estoy un poco confundido de si es un problema del mismo Anki o de Yomichan ya que el problema cambió al moverme a las versiones 2.15.0 y 2.15.1 de Ankidroid donde se soluciona el problema del botón "+" (que no aparecía) y me permitió agregar las tarjetas, sin embargo, puedo agregarlas de manera infinita ya que el pequeño libro se sustituye por el botón de "+" que no desaparece por mas que las agregue.

![Screenshot_20221001-173547_Kiwi Browser](https://user-images.githubusercontent.com/114822001/193429991-ae65a5be-1189-48d4-87b4-af3ca6f3e6e8.jpg)
![Screenshot_20221001-173710_Kiwi Browser](https://user-images.githubusercontent.com/114822001/193429992-b2c655fa-f199-4010-9ffd-c56519c0c98f.jpg)

**Comment by toasted-nutbread** [comment]
Google translated:

> I can't add most of the words I select in Yomichan to Ankidroid for two days (the "+" button is not showing). Similarly, the words that I already have added do not show the little book that indicates that they have already been sent to Anki.
> 
> The truth is that I am a bit confused if it is a problem with Anki itself or with Yomichan since the problem changed when I moved to versions 2.15.0 and 2.15.1 of Ankidroid where the problem with the "+" button (which did not appear ) and it allowed me to add the cards, however, I can add them infinitely since the little book is replaced by the "+" button that does not disappear no matter how many times I add them.

Can you share and export your Yomichan setting file?

![image](https://user-images.githubusercontent.com/11037431/193435369-98adbb29-614c-4697-ad3e-e734f5ac9412.png)

**Comment by Patata099** [comment]
> Google tradujo:
> 
> > No puedo agregar la mayoría de las palabras que selecciono en Yomichan a Ankidroid durante dos días (el botón "+" no aparece). Del mismo modo, las palabras que ya he agregado no muestran el librito que indica que ya fueron enviadas a Anki.
> > La verdad es que estoy un poco confundido si es un problema con el mismo Anki o con Yomichan ya que el problema cambió cuando pasé a las versiones 2.15.0 y 2.15.1 de Ankidroid donde el problema con el botón "+" (que no no aparece) y me permitía agregar las tarjetas, sin embargo, puedo agregarlas infinitamente ya que el librito se reemplaza por el botón "+" que no desaparece por más veces que las agregue.
> 
> ¿Puedes compartir y exportar tu archivo de configuración de Yomichan?
> 
> ![imagen](https://user-images.githubusercontent.com/11037431/193435369-98adbb29-614c-4697-ad3e-e734f5ac9412.png)

¿Cómo puedo enviartelo? ¿debería enviarte lo que contiene el archivo resultante?

**Comment by Patata099** [comment]
file:///storage/emulated/0/Y/

**Comment by toasted-nutbread** [comment]
Rename the file to a `.txt` instead of a `.json`, and then you can attach it to a comment.

Google translated:

> Cambie el nombre del archivo a `.txt` en lugar de `.json`, y luego puede adjuntarlo a un comentario.

**Comment by Patata099** [comment]
[yomichan-settings-2022-10-02-02-38-53.txt](https://github.com/FooSoft/yomichan/files/9692892/yomichan-settings-2022-10-02-02-38-53.txt)

**Comment by landonepps** [comment]
I'm experiencing the same issue on Linux. If I disable checking for duplicates, the + button always appears. But on some words that were missing the + button, I get an error saying:
```
An error occurred:

    Anki error: cannot create note because it is empty

Log debug info to console
```
This happens for the entry にもかかわらず, for instance.
[yomichan-settings-2022-10-02-15-48-10.txt](https://github.com/FooSoft/ymoichan/files/9692909/yomichan-settings-2022-10-02-15-48-10.txt)

**Comment by landonepps** [comment]
Ah, I think I figured it out when rereading the error log.
My sort field is Optimized-Voc-Index which I set to the frequency. And I'm guessing the sort field is required by Anki, and the words I'm trying to add has no frequency info.

**Comment by toasted-nutbread** [comment]
@landonepps It looks like the first field of the card is `Optimized-Voc-Index`, which is empty. The first field must be unique and non-empty, since this is what Anki uses for uniqueness checks.

**Comment by landonepps** [comment]
Makes sense. I wonder if Patata099 has the same issue. The field may either be blank or not unique.
It would be nice to update the error message to say something about this requirement.

Changing “cannot create note because it is empty“ to “cannot create note because the first field is empty” would help a lot with understanding the actual issue.

Though the error only appears if you disable duplicate checks, which is hidden in the advanced settings, so it’s possible most people will never discover the error message to begin with. They’ll just be confused why the + button doesn’t appear.

**Comment by toasted-nutbread** [comment]
@Patata099 Check your Anki card format under _Configure Anki card format..._ and share a screenshot if you can.

I've seen several people use a model named "Mining-475ae", which yours does as well, but the format doesn't match the format I've seen used in the past. If you have fields marked in red, they are invalid. Example:

![image](https://user-images.githubusercontent.com/11037431/193959571-925ef09b-d1ab-4370-b3c1-1396ed449f3a.png)

**Comment by toasted-nutbread** [comment]
@landonepps 
> though I imagine that would be rather difficult since the templates can be customized.

This is correct, there wouldn't be a way to easily and correctly determine if the first field is valid. Some of the Anki card format errors could be better indicated I suppose, with a description about the first field.

**Comment by Patata099** [comment]
> @Patata099Verifica el formato de tu tarjeta Anki en _Configurar formato de tarjeta Anki..._ y comparte una captura de pantalla si puedes.
> 
> He visto a varias personas usar un modelo llamado "Mining-475ae", que el tuyo también usa, pero el formato no coincide con el formato que he visto usar en el pasado. Si tiene campos marcados en rojo, no son válidos. Ejemplo:
> 
> ![imagen](https://user-images.githubusercontent.com/11037431/193959571-925ef09b-d1ab-4370-b3c1-1396ed449f3a.png)

Aquí lo dejo. 

![Screenshot_20221004-213517_Kiwi Browser](https://user-images.githubusercontent.com/114822001/193961164-4417a7e3-b5a1-4aab-83e5-0b660a030237.jpg)

En cuanto al por qué de que sea diferente podría ser que le hice algunos cambios como por el ejemplo agregar el pitch accent en " audio de la oración ". No me apareció nada en rojo asi que lo dejé. No sé si ahí estará el problema.

**Comment by toasted-nutbread** [comment]
Was the image taken while connected to Anki? _Enable Anki integration_ would show this:

![image](https://user-images.githubusercontent.com/11037431/194184571-d4a92c26-cbd3-4084-b737-76367e68810a.png)

**Comment by Patata099** [comment]
> ¿Se tomó la imagen mientras estaba conectado a Anki? _Habilitar la integración de Anki_ mostraría esto:
> 
> ![imagen](https://user-images.githubusercontent.com/11037431/194184571-d4a92c26-cbd3-4084-b737-76367e68810a.png)

En todas las imágenes que te he enviado, sí.

**Comment by toasted-nutbread** [comment]
What type of device are you running Anki and Yomichan on? And what web browser are you using?

**Comment by Patata099** [comment]
> ¿En qué tipo de dispositivo estás ejecutando Anki y Yomichan? ¿Y qué navegador web estás usando?

Kiwi Browser

![20221005_201749](https://user-images.githubusercontent.com/114822001/194186466-2eaf62eb-fe34-433a-9897-e36f0aab3e54.jpg)

**Comment by toasted-nutbread** [comment]
How are you using AnkiConnect on Android?

**Comment by Patata099** [comment]
> ¿Cómo estás usando AnkiConnect en Android?

Debido a que en Ankidroid no existen los complementos, AnkiConnet en android es una app. Sólo necesita encenderlo en "start" y ya conecta directamente con Anki.
![Uploading Screenshot_20221005-210036_Ankiconnect Android.jpg…]()

**Comment by toasted-nutbread** [comment]
Where does this app come from? I cannot find it with some cursory searching.

**Comment by Patata099** [comment]
> ¿De dónde viene esta aplicación? No puedo encontrarlo con una búsqueda superficial.

Si no recuerdo mal creo que aquí la obtuve.

https://github.com/KamWithK/AnkiconnectAndroid

**Comment by toasted-nutbread** [comment]
It's possible that the issue is with either that app or Ankidroid; that app isn't something I've used before so it's not something we target in our testing.

Your settings seem to work as expected on a normal desktop configuration.

Also see https://github.com/KamWithK/AnkiconnectAndroid/blob/master/README.md#limitations, as this seems related to your issue.

**Comment by Patata099** [comment]
> Es posible que el problema sea con esa aplicación o con Ankidroid; esa aplicación no es algo que haya usado antes, por lo que no es algo a lo que apuntamos en nuestras pruebas.
> 
> Su configuración parece funcionar como se esperaba en una configuración de escritorio normal.
> 
> Consulte también https://github.com/KamWithK/AnkiconnectAndroid/blob/master/README.md#limitations , ya que parece estar relacionado con su problema.

Ya veo. Sin duda saber de donde proviene el problema es de gran ayuda. ¡Muchas gracias!

**Comment by Aquafina-water-bottle** [comment]
@Patata099 I believe the bug is fixed now. Try using the latest version of AnkiconnectAndroid: https://github.com/KamWithK/AnkiconnectAndroid/releases/tag/1.6-beta (you likely have to remove the existing app and re-install the APK).
If you still have problems, I'd recommend closing this issue and opening a new issue in the AnkiconnectAndroid repository linked above.
